src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -35,49 +122,59 @@ private RuntimeOSPlatformInfo(string invokedPlatformCheckMethodName, string plat
             public bool Negated { get; }
 
             public IAbstractAnalysisValue GetNegatedValue()
-                => new RuntimeOSPlatformInfo(InvokedPlatformCheckMethodName, PlatformPropertyName, Version, !Negated);
+                => new RuntimeMethodValue(InvokedPlatformCheckMethodName, PlatformPropertyName, Version, !Negated);
 
             public static bool TryDecode(
                 IMethodSymbol invokedPlatformCheckMethod,
                 ImmutableArray<IArgumentOperation> arguments,
                 ValueContentAnalysisResult? valueContentAnalysisResult,
                 INamedTypeSymbol osPlatformType,
-                [NotNullWhen(returnValue: true)] out RuntimeOSPlatformInfo? info)
+                [NotNullWhen(returnValue: true)] out RuntimeMethodValue? info)
             {
-                if (!TryDecodeOSPlatform(arguments, osPlatformType, out var osPlatformProperty) ||
+                Debug.Assert(!arguments.IsEmpty);
+
+                if (arguments[0].Value is ILiteralOperation literal && literal.Type.SpecialType == SpecialType.System_String)
```

Comment:
Protect against null literal.

```suggestion
                if (arguments[0].Value is ILiteralOperation literal && literal.Type?.SpecialType == SpecialType.System_String)
```